forum/qa-plugin/q2a-badges-master/qa-badge-check.php

```diff
@@ -358,20 +358,17 @@ function answer_vote_down($event,$event_user,$params) {
 			if($event_user) $this->check_voter($event_user);
 		}
 
-		function check_voted($uid) {
-						// upvotes received
+                function check_voted($uid) {
+                    // upvotes received
 			
-			$votes = qa_db_read_one_assoc(
-				qa_db_query_sub(
-					'SELECT upvoteds FROM ^userpoints WHERE userid=#',
-					$uid
-				),
-				true
-			);
-			$badges = array('liked','loved','revered');
+                    $votesArray = qa_db_query_raw('SELECT upvoteds FROM qa_userpoints WHERE userid='.$uid.' LIMIT 1');
+                    foreach($votesArray as $vote) {
+                        $votes = $vote;

```

Comment:
Średnio rozumiem jaki sens ma tutaj robienie pętli, której każde przejście nadpisuje jedną zmienną nową wartością.

Reply:
będzie tylko jedno przejście (LIMIT  1)

Reply:
To w takim razie nie rozumiem po co pętla :p